File: src/main/java/com/footverse/product/dto/CreateProductVariantRequest.java
package com.footverse.product.dto;

import java.math.BigDecimal;

import com.footverse.product.entity.ProductVariantStatus;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.Size;

/**
 * Admin request to create a product variant (validation-spec §12). The {@code (color, size)} pair
 * must be unique within the product.
 *
 * @param color         the variant color; required; at most 50 characters
 * @param size          the variant size; required
 * @param stockQuantity the stock on hand; required; zero or positive
 * @param sku           the stock-keeping unit; required; unique
 * @param priceOverride the optional price override; when null the product's base price applies
 * @param status        the variant status; required
 */
public record CreateProductVariantRequest(
        @NotBlank @Size(max = 50) String color,
        @NotBlank String size,
        @NotNull @Min(0) Integer stockQuantity,
        @NotBlank String sku,
        @Positive BigDecimal priceOverride,
        @NotNull ProductVariantStatus status) {
}

File: admin/src/features/product/models/create-product-variant-request.ts
import type { ProductVariantStatus } from './product-variant-status';

/**
 * Create a product variant (dto-spec §9). Mirrors the frozen DTO
 * field-for-field, including the ADMIN-only `costPrice` (Sprint 11).
 */
export interface CreateProductVariantRequest {
  readonly color: string;
  readonly size: string;
  readonly stockQuantity: number;
  readonly sku: string;
  readonly priceOverride?: number;
  readonly status: ProductVariantStatus;
  readonly costPrice: number;
}
